homeassistant/components/zwave_js/discovery.py

```diff
@@ -263,25 +424,43 @@ def async_discover_values(node: ZwaveNode) -> Generator[ZwaveDiscoveryInfo, None
                 and value.node.device_class.specific not in schema.device_class_specific
             ):
                 continue
-            # check command_class
-            if (
-                schema.command_class is not None
-                and value.command_class not in schema.command_class
-            ):
-                continue
-            # check endpoint
-            if schema.endpoint is not None and value.endpoint not in schema.endpoint:
-                continue
-            # check property
-            if schema.property is not None and value.property_ not in schema.property:
-                continue
-            # check metadata_type
-            if schema.type is not None and value.metadata.type not in schema.type:
+            # check primary value
+            if not check_value(value, schema.primary_value):
                 continue
+            # check additional required values
+            if schema.required_values is not None:
+                pass
+
             # all checks passed, this value belongs to an entity
             yield ZwaveDiscoveryInfo(
                 node=value.node,
                 primary_value=value,
                 platform=schema.platform,
                 platform_hint=schema.hint,
             )
+            if not schema.allow_multi:
+                # break out of loop, this value may not be discovered by other schemas/platforms
+                break
+
+
+@callback
+def check_value(value: ZwaveValue, schema: ZWaveValueDiscoverySchema) -> bool:
+    """Check if value matches scheme."""
+    if value is None or schema is None:
```

Comment:
This can't happen according to the typing. It would be good to check this case before calling this function if it can happen.

Reply:
It can't happen indeed. Removed it.